### ISSUE TITLE
Macros not firing in 10.4.0

### ISSUE THREAD

**Issue by mawdesley** [architect/architect#1364]
**Describe the issue**
Macros are no longer called on 10.4.0

**Steps to reproduce**
I've made an example project here: https://github.com/mawdesley/arc-macro-test , it's just the initial template +  a macro which prints "MACRO CALLED!!!" to the console.

1. clone the repo
2. `npm i`
3. you may need to alter the aws profile to something that works for you - I've set it to "personal"
4.  `npx arc deploy --dry-run`
5.  observe that "MACRO CALLED!!!" is not printed to the console

**Expected behavior**
Macros should be called on deploy

The example repo ^^^ has a previous commit which is pegged to arc 10.3.4. You can see the expected behaviour by:

1. removing package-lock.json & node modules
2. `git checkout HEAD^`
3. `npm i`
4. `npx arc deploy --dry-run`
5. Observe "MACRO CALLED!!!" printed to the console

**Screenshots**
10.4.0:
![Screenshot from 2022-08-05 10-20-04](https://user-images.githubusercontent.com/44057041/183049305-29ad2517-dbec-4a05-9ac4-5e4d7614ee3b.png)

10.3.4:
![image](https://user-images.githubusercontent.com/44057041/183049791-faa0cdb2-e9b1-4ba6-9636-43be3db44db5.png)

**Desktop**
Please complete the following information (if appropriate):
 - OS: Ubuntu 22.04

**Comment by lpsinger** [comment]
Confirmed that deploy.start plugins are not being fired either in architect 10.4.

**Comment by ryanblock** [comment]
Oof, sorry folks! Tiny internal change that I didn't account for in that path, fixed here: https://github.com/architect/deploy/pull/358/commits/119299def066c3e4dab67d147e565207780628ba. Patch will be released in a few mins as 10.4.1.

**Comment by ryanblock** [comment]
For future reference, how this happened: I made an internal change to Inventory when working on the new `plugin.set` APIs because we added some new behavior (and a new property) to allow the project manifest to override plugins. This internal change was not reflected in Deploy, and tests did not catch it because they utilized a stripped down mock instead of an actual Inventory run of a project (see: https://github.com/architect/deploy/commit/119299def066c3e4dab67d147e565207780628ba#diff-2012e3de2b7011b1956ed644d728cc1ba9ff885dd50b259fb7b99c17a37ed923).

My manual testing / verification setup, which I did run before shipping this release, happened to not have any deploy.start plugins, so I didn't catch it there either. Pretty disappointing bug to slip through, hope that didn't ruin anyone's Friday!
